gnovm/pkg/gnolang/preprocess.go

```diff
@@ -882,17 +882,22 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 					// Left not const, Right not const ------------------
 					if n.Op == EQL || n.Op == NEQ {
 						// If == or !=, no conversions.
-					} else if lnt, ok := lt.(*NativeType); ok {
+					} else if lnt, ok := lt.(*NativeType); ok && isNative(rt) {
```

Comment:
this seems wrong. when Lhs is native and Rhs not native, the logic inside the `else if` block is skipped, and the Lhs will not be converted to gno type, so the left type and right type will never match.. right?

Reply:
I believe that when Lhs is native and Rhs is not native, they are not of the same type, which should result in an error being raised.

Referring to the comment at https://github.com/gnolang/gno/blob/master/gnovm/pkg/gnolang/preprocess.go#L891-L892, where we write that we should only enter the 'else if' condition when lt is native and rt is Native. Do you have any scenarios in mind where lt is native and rt is not native, but we should not raise an error?

Reply:
make sense to me, ty.